bundles/best_practices.rst

```diff
@@ -546,6 +546,78 @@ Beware that templates use a simplified version of the logical path shown above.
 For example, an ``index.html.twig`` template located in the ``templates/Default/``
 directory of the AcmeBlogBundle, is referenced as ``@AcmeBlog/Default/index.html.twig``.
 
+Specifics for a UX Bundle
+-------------------------
+
+Here are a few tricks to make your bundle install as a UX bundle.
+
+composer.json file
+~~~~~~~~~~~~~~~~~~
+
+Your ``composer.json`` file must have the ``symfony-ux`` keyword:
+
+.. code-block:: json
+
+    {
+        "keywords": ["symfony-ux"]
+    }
+
+Assets location
+~~~~~~~~~~~~~~~
+
+Your assets must be located in one of the following directories, with a ``package.json`` file so Flex can handle it
+during install/update:
+
+* ``/assets``
+* ``/Resources/assets``
+* ``/src/Resources/assets``
+
+package.json file
+~~~~~~~~~~~~~~~~~
+
+Your ``package.json`` file must contain a ``symfony`` config with controllers defined, and also add required packages
+to the ``peerDependencies``:
+
+.. code-block:: json
+
+    {
+        "name": "@acme/feature",
+        "symfony": {
+            "controllers": {
+                "slug": {
+                    "main": "dist/controller.js",
+                    "webpackMode": "eager",
+                    "fetch": "eager",
+                    "enabled": true,
+                    "autoimport": {
+                        "dist/bootstrap4-theme.css": false,
+                        "dist/bootstrap5-theme.css": true
+                    }
+                }
+            }
+        },
+        "peerDependencies": {
+            "@hotwired/stimulus": "^3.0.0",
+            "slugify": "^1.6.5"
+        }
+    }
+
+In this case, the file located at ``[assets directory]/dist/controller.js`` will be exposed.
+
+If the bundle is named ``acme/feature``, the base key for the controller will be ``acme--feature--slug``: it's composed
+of the bundle name and the controller name defined in the ``package.json`` file.
+
+To use it in a template (e.g. one defined in your bundle) you can use it like this:
+
+.. code-block:: html+twig
+
+    <div
+        data-controller="acme--feature--slug"
+        data-acme--feature--slug-modal-value="my-value"
+    >
+        ...
+    </div>
```

Comment:
What about some info about building the `dist/` directory?

Reply:
This can vary a lot, based on developers and their config. Here's what I can tell:

* building using Encore (but AFAICT noone does that) is kinda hard when it's for a bundle (see e.g. issues with Easyadmin)
* build tools & configs are not the same if you use Sass or not, Typescript or not, if you use babel or an other transpiler, and so on
* I did browse a few ux packages (using Packagist: https://packagist.org/?tags=symfony%20ux) and most of them point directly to a raw .js file that's not transpiled or compiled or anything)

Reply:
What if we outline 2 main recommendations:

A) Write raw JS that won't be transpiled. In this case, it should live in the `dist/` directory since that's where the `package.json` is pointing
B) Use TypeScript. And then it'd be great to give some basic info about how to set that up. Or, perhaps we need to somehow make this easier for people - the setup in UX is quite complex. The goal should be to enable people to use TS and have a really nice setup without much work or knowledge.

Reply:
Done :)